collect_app/src/main/java/org/odk/collect/android/dao/InstancesDao.java

```diff
@@ -145,6 +146,41 @@ public void deleteInstancesDatabase() {
         Collect.getInstance().getContentResolver().delete(InstanceProviderAPI.InstanceColumns.CONTENT_URI, null, null);
     }
 
+    public void deleteInstancesFromIDs(String ids[]){
+        int count = ids.length;
+        int i = 0 ;
```

Comment:
I think the code will work as it is.

If you want tighter code, instead of looping and incrementing i for every element, you could use counter here. And then reference of the `ids[i]` will become `ids[counter * ApplicationConstants.SQLITE_MAX_VARIABLE_NUMBER + j]`.

The pros:
* No looping to increment i.

The cons:
* Might make it less understandable.